shelley/chain-and-ledger/formal-spec/epoch.tex

```diff
@@ -810,6 +825,31 @@ \subsection{Complete Epoch Boundary Transition}
 
 The epoch transition rule calls $\mathsf{SNAP}$, $\mathsf{POOLREAP}$ and $\mathsf{NEWPP}$
 in sequence.
+Figure~\ref{fig:funcs:epoch-helper} gives some helper functions for the
+$\mathsf{EPOCH}$ transition.
+
+%%
+%% Figure - Helper Functions for Epoch Rules
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{pvotes} \in \Update \to \Ppm \to (\VKeyGen \mapsto \Value) \\
+      & \fun{pvotes}~\var{up}~\var{ppm}
+      = \{k\mapsto v~\mid~ k\mapsto\var{ppm} ~\land~ \var{ppm}\mapsto v\}\\
+  \end{align*}
+
+  \begin{align*}
+      & \fun{tally} \in \Update \to \PParams\\
+      & \fun{tally}~\var{up} =
+          \Big\{\var{ppm}\mapsto v
+          ~\Big|~
+          |\fun{pvotes}~\var{up}~\var{ppm}\restrictrange\var{v}|\geq 5 \Big\}
+  \end{align*}
+
```

Comment:
I don't know if `tally` is well defined in the case that you have a protocol parameter with two majorities (which if we have 7 keys shouldn't happen, but still the definition of `tally` doesn't now this ;) )

Reply:
I'm making the assumption that there are only ever seven keys. maybe I can state that somewhere.